importer/src/main/java/com/example/akka/directdebit/importer/application/ListFilesWorkflow.java

```diff
@@ -126,6 +126,9 @@ public WorkflowDef<FileListState> definition() {
           Source.from(currentState().files())
                   .mapAsync(currentState().files().size(),
                           file -> componentClient.forWorkflow(WorkflowId.fileImportWorkflowId(file.fileName(),currentState().folder()))
+                                                   // COMMENT: just a heads-up that there is a limit on the number of
+                                                   // concurrent workflows. Make sure you don't exceed it during a
+                                                   // demo. Otherwise, you can throttle the workflow creation.
```

Comment:
I think that in general, the problem we have here is that we are treating the domain model separately from the entity/workflow. The result is that we need to return types that express success and failure and then convert them to the Effect API. 

We need to spend some more time on this kind of code ergonomics. For this project it's fine because its goal is to demo the SDK. However, we need to iterate a bit more on this issue to have a more streamline recommendation and then show case it in samples/quickstarts.

Reply:
the domain can return all the combinations of events/no events and error/no error.
But I agree we need to work on this code ergonomics.